app/controllers/util/FormField.scala

```diff
@@ -210,14 +211,12 @@ object TextField {
 
   def validate(text: String, textField: TextField): Option[String] =
     if (text == "") {
-      val fieldType: String = inputModeToFieldType(textField.inputMode)
-      val fieldName = if (textField.fieldId == "TitleAlternate") "alternative title" else textField.fieldName
-      Some(emptyValueError.format(fieldType, fieldName))
+      val fieldName = if (textField.fieldAlternativeName.isEmpty) textField.fieldName.toLowerCase else textField.fieldAlternativeName
+      if (textField.fieldId == "ClosurePeriod") Some(closurePeriodNotEnteredError.format(fieldName)) else Some(dependencyNotEntered.format(fieldName))
     } else if (textField.inputMode.equals("numeric")) {
-      val inputName = textField.nameAndValue.name
       text match {
-        case t if allCatch.opt(t.toInt).isEmpty => Some(numberError.format(inputName))
-        case t if t.toInt < 0                   => Some(negativeNumberError.format(inputName))
+        case t if allCatch.opt(t.toInt).isEmpty => Some(wholeNumberError.format(textField.fieldName.toLowerCase, "3, 15, 21"))
```

Comment:
The allCatch thing is nice, not seen that before :+1:

Reply:
Could you put the "3, 15, 21" in a variable. It's in here a couple of times and it will make it easier to update.